src/Http/Http.Api/src/MapActionExpressionTreeBuilder.cs

```diff
@@ -0,0 +1,385 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http.Api.Abstractions;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Http.Api
+{
+    // Borrows heavily from https://github.com/davidfowl/uController/blob/aa8bcb4b30764e42bd72d326cb2718a4d4eaf4a9/src/uController/HttpHandlerBuilder.cs
```

Comment:
Hmmmm, I don't think we need this.

Reply:
@halter73 and I discussed turning this in to a source generator at some point. It's not necessary for a preview (or even a v1), but it would be cool if we can provably use one before we ship a RTM.

Reply:
I'll remove this. It left it mostly as a note for reviewers.